### ISSUE TITLE
[question] How to build for Mac Catalyst?

### ISSUE THREAD

**Issue by dmn-star** [conan-io/conan#8388]
1.33 with catalyst support is out and I tried it right away. Maybe with a bit naive first try. :-)

### Environment Details (include every applicable attribute)
  * Operating System+version: Catalina
  * Compiler+version: Apple clang version 12.0.0 (clang-1200.0.32.28)
  * Conan version: 1.33
  * Python version:  3.7.4

### Steps to reproduce (Include if Applicable)
conan install xmlsec/1.2.31@ --build --profile ./conan/profiles/mac_catalyst

### Error
from config.log

>configure:4198: $? = 0
configure:4187: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -V >&5
clang: error: argument to '-V' is missing (expected 1 value)
clang: error: no input files
configure:4198: $? = 1
configure:4187: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -qversion >&5
**clang: error: unknown argument '-qversion'; did you mean '--version'?**
clang: error: no input files
configure:4198: $? = 1
configure:4218: checking whether the C compiler works
configure:4240: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -target=x86_64-apple-ios-macabi -O3 --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fPIC -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -DNDEBUG -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -target=x86_64-apple-ios-macabi --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 conftest.c  >&5
**clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?
clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?**
configure:4244: $? = 1

______
???
configure:4240: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang 
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64
-target=x86_64-apple-ios-macabi
-O3
--sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-fPIC 
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64 -DNDEBUG
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk 
-arch x86_64
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64 -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-arch x86_64
-target=x86_64-apple-ios-macabi
--sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk
-isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk 
-arch x86_64 conftest.c  >&5

### Profile
>[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=12.0
os=Macos
os.subsystem=Catalyst
os_build=Macos
[options]
[build_requires]
*: darwin-toolchain/1.0.8@theodelrieu/stable
[env]

### Logs (Executed commands with output) (Include/Attach if Applicable)

```
conan install xmlsec/1.2.31@ --build --profile ./conan/profiles/mac_catalyst
Configuration:
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.cppstd=14
compiler.libcxx=libc++
compiler.version=12.0
os=Macos
os.subsystem=Catalyst
os_build=Macos
[options]
[build_requires]
*: darwin-toolchain/1.0.8@theodelrieu/stable
[env]

libiconv/1.16: Forced build from source
openssl/1.1.1h: Forced build from source
zlib/1.2.11: Forced build from source
libxml2/2.9.10: Forced build from source
xmlsec/1.2.31: Forced build from source
darwin-toolchain/1.0.8@theodelrieu/stable: Forced build from source
m4/1.4.18: Forced build from source
pkgconf/1.7.3: Forced build from source
autoconf/2.69: Forced build from source
automake/1.16.3: Forced build from source
libtool/2.4.6: Forced build from source
ninja/1.10.2: Forced build from source
meson/0.55.1: Forced build from source
Installing package: xmlsec/1.2.31
Requirements
    libiconv/1.16 from 'conan-center' - Cache
    libxml2/2.9.10 from 'conan-center' - Cache
    openssl/1.1.1h from 'conan-center' - Cache
    xmlsec/1.2.31 from 'conan-center' - Cache
    zlib/1.2.11 from 'conan-center' - Cache
Packages
    libiconv/1.16:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
    libxml2/2.9.10:a2fcfc5995ddeefb1ff988dbc9aa90b909d43ed5 - Build
    openssl/1.1.1h:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
    xmlsec/1.2.31:6c9b5886a0769be2cb726e1b2e87b7bc5c550492 - Build
    zlib/1.2.11:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build
Build requirements
    autoconf/2.69 from 'conan-center' - Cache
    automake/1.16.3 from 'conan-center' - Cache
    darwin-toolchain/1.0.8@theodelrieu/stable from 'conan-center' - Cache
    libtool/2.4.6 from 'conan-center' - Cache
    m4/1.4.18 from 'conan-center' - Cache
    meson/0.55.1 from 'conan-center' - Cache
    ninja/1.10.2 from 'conan-center' - Cache
    pkgconf/1.7.3 from 'conan-center' - Cache
Build requirements packages
    autoconf/2.69:923c5310a94256af03d94c969eebbbadac513c42 - Build
    automake/1.16.3:8198ae767c7bf455a38775936906b3e982c035ec - Build
    darwin-toolchain/1.0.8@theodelrieu/stable:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build
    libtool/2.4.6:072e844e71af4f1016fea59345350a58177d9979 - Build
    m4/1.4.18:3bc8a0eb26ab34ed4958480d81dfd67c24f74ed4 - Build
    meson/0.55.1:5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9 - Build
    ninja/1.10.2:3f3351f8c9c8848e68a09e5f612a66c58eb79473 - Build
    pkgconf/1.7.3:9d3239a97b703c82f38d9e91676ca971d7eb7701 - Build

Installing (downloading, building) binaries...
darwin-toolchain/1.0.8@theodelrieu/stable: Copying sources to build folder
darwin-toolchain/1.0.8@theodelrieu/stable: Building your package in /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] 'fPIC' option not found
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
darwin-toolchain/1.0.8@theodelrieu/stable: Calling build()
darwin-toolchain/1.0.8@theodelrieu/stable: WARN: This conanfile has no build step
darwin-toolchain/1.0.8@theodelrieu/stable: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' built
darwin-toolchain/1.0.8@theodelrieu/stable: Build folder /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/build/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Generated conaninfo.txt
darwin-toolchain/1.0.8@theodelrieu/stable: Generated conanbuildinfo.txt
darwin-toolchain/1.0.8@theodelrieu/stable: Generating the package
darwin-toolchain/1.0.8@theodelrieu/stable: Package folder /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9
darwin-toolchain/1.0.8@theodelrieu/stable: Calling package()
[HOOK - conan-center.py] post_package(): ERROR: [PACKAGE LICENSE (KB-H012)] No 'licenses' folder found in package: /Users/dmn/.conan/data/darwin-toolchain/1.0.8/theodelrieu/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9  (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H012)
[HOOK - conan-center.py] post_package(): ERROR: [DEFAULT PACKAGE LAYOUT (KB-H013)] Unknown file 'darwin-toolchain.cmake' in the package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H013)
[HOOK - conan-center.py] post_package(): [DEFAULT PACKAGE LAYOUT (KB-H013)] If you are trying to package a tool put all the contents under the 'bin' folder
[HOOK - conan-center.py] post_package(): ERROR: [MATCHING CONFIGURATION (KB-H014)] Empty package (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H014)
[HOOK - conan-center.py] post_package(): ERROR: [MATCHING CONFIGURATION (KB-H014)] Packaged artifacts does not match the settings used: os=Macos, compiler=apple-clang (https://github.com/conan-io/conan-center-index/blob/master/docs/error_knowledge_base.md#KB-H014)
[HOOK - conan-center.py] post_package(): [SHARED ARTIFACTS (KB-H015)] OK
[HOOK - conan-center.py] post_package(): [PC-FILES (KB-H020)] OK
[HOOK - conan-center.py] post_package(): [CMAKE-MODULES-CONFIG-FILES (KB-H016)] OK
[HOOK - conan-center.py] post_package(): [PDB FILES NOT ALLOWED (KB-H017)] OK
[HOOK - conan-center.py] post_package(): [LIBTOOL FILES PRESENCE (KB-H018)] OK
[HOOK - conan-center.py] post_package(): [MS RUNTIME FILES (KB-H021)] OK
darwin-toolchain/1.0.8@theodelrieu/stable package(): Packaged 1 '.cmake' file: darwin-toolchain.cmake
darwin-toolchain/1.0.8@theodelrieu/stable: Package '5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9' created
darwin-toolchain/1.0.8@theodelrieu/stable: Created package revision c093366e06ad120604635abd2f99a5d2
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
[HOOK - conan-center.py] post_package_info(): [CMAKE FILE NOT IN BUILD FOLDERS (KB-H019)] OK
libiconv/1.16: Applying build-requirement: darwin-toolchain/1.0.8@theodelrieu/stable
libiconv/1.16: WARN: Build folder is dirty, removing it: /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
libiconv/1.16: Copying sources to build folder
libiconv/1.16: Building your package in /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
libiconv/1.16: Generator txt created conanbuildinfo.txt
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK. 'fPIC' option found and apparently well managed
[HOOK - conan-center.py] pre_build(): [FPIC MANAGEMENT (KB-H007)] OK
libiconv/1.16: Calling build()
libiconv/1.16: Calling:
 > ./configure '--enable-static' '--disable-shared' '--prefix=/Users/dmn/.conan/data/libiconv/1.16/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share'
checking for a BSD-compatible install... /usr/local/bin/ginstall -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/local/bin/gmkdir -p
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make sets $(MAKE)... (cached) yes
checking for gcc... /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang
checking whether the C compiler works... no
configure: error: in `/Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701/source_subfolder':
configure: error: C compiler cannot create executables
See `config.log' for more details
libiconv/1.16:
libiconv/1.16: ERROR: Package '9d3239a97b703c82f38d9e91676ca971d7eb7701' build failed
libiconv/1.16: WARN: Build folder /Users/dmn/.conan/data/libiconv/1.16/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
ERROR: libiconv/1.16: Error in build() method, line 127
	autotools = self._configure_autotools()
while calling '_configure_autotools', line 117
	self._autotools.configure(args=configure_args, host=host, build=build)
	ConanException: Error 77 while executing ./configure '--enable-static' '--disable-shared' '--prefix=/Users/dmn/.conan/data/libiconv/1.16/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--bindir=${prefix}/bin' '--sbindir=${prefix}/bin' '--libexecdir=${prefix}/bin' '--libdir=${prefix}/lib' '--includedir=${prefix}/include' '--oldincludedir=${prefix}/include' '--datarootdir=${prefix}/share'
```

**Comment by dmn-star** [comment]
@SSE4 What all do I have to consider or is this now an error in Catalyst support?

**Comment by memsharded** [comment]
Asking @SSE4 for help.

It would be great to know if it builds and works for simpler use cases, like typical "hello world", or even other Conan packages like zlib or others. Maybe it is something of that recipe in particular, that its build system is nor prepared to build for Apple Catalyst.

**Comment by memsharded** [comment]
If it is a bug @SSE4, it can be directed to 1.33.1.

**Comment by dmn-star** [comment]
I get the same error for Zlib too.

>zlib/1.2.11: Calling build()
zlib/1.2.11: Calling:
../configure '--prefix=/Users/dmn/.conan/data/zlib/1.2.11/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--libdir=${prefix}/lib' '--includedir=${prefix}/include'
Compiler error reporting is too harsh for ../configure (perhaps remove -Werror).
./configure aborting.
zlib/1.2.11:
zlib/1.2.11: ERROR: Package '9d3239a97b703c82f38d9e91676ca971d7eb7701' build failed
zlib/1.2.11: WARN: Build folder /Users/dmn-star/.conan/data/zlib/1.2.11/_/_/build/9d3239a97b703c82f38d9e91676ca971d7eb7701
ERROR: zlib/1.2.11: Error in build() method, line 41
	self._build_zlib()
while calling '_build_zlib', line 111
	self._build_zlib_autotools()
while calling '_build_zlib_autotools', line 76
	env_build.configure("../", build=False, host=False, target=False)
	ConanException: Error 1 while executing ../configure '--prefix=/Users/dmn-star/.conan/data/zlib/1.2.11/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701' '--libdir=${prefix}/lib' '--includedir=${prefix}/include'

________

.../configure --prefix=/Users/dmn-star/.conan/data/zlib/1.2.11/_/_/package/9d3239a97b703c82f38d9e91676ca971d7eb7701 --libdir=${prefix}/lib --includedir=${prefix}/include
Fr 22 Jan 2021 23:06:16 CET
=== ztest70324.c ===
extern int getchar();
int hello() {return getchar();}

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -c ztest70324.c
... using gcc

Checking for obsessive-compulsive compiler options...
=== ztest70324.c ===
int foo() { return 0; }

/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -c -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 -target=x86_64-apple-ios-macabi -O3 --sysroot=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -fPIC -isysroot/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 ztest70324.c
**clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?
(exit code 1)**
Compiler error reporting is too harsh for ../configure (perhaps remove -Werror).
../configure aborting.

**Comment by SSE4** [comment]
the error is:
```
clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?
(exit code 1)
```
that's curious, as several people stated it should be `-target`.

**Comment by SSE4** [comment]
confirm the same error with `zlib/1.2.11`:
```
clang -c -target=x86_64-apple-ios-macabi -O3 -fPIC -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -arch x86_64 ztest99700.c
clang: error: unknown argument '-target=x86_64-apple-ios-macabi'; did you mean '--target=x86_64-apple-ios-macabi'?
(exit code 1)
```

**Comment by SSE4** [comment]
changing to `--target=x86_64-apple-ios-macabi`, `zlib/1.2.11` build successfully. `openssl/1.1.1i` also builds fine.
gtest complains:
```
error: 'pthread_key_create' is only available on macCatalyst 13.0 or newer
      [-Werror,-Wunguarded-availability-new]
        pthread_key_create(&key, &DeleteThreadLocalValue));
        ^~~~~~~~~~~~~~~~~~
```
adding `os.version=13.0` to profile it fails on later stage:
```
clang: error: overriding '-mmacosx-version-min=13.0' option with '--target=x86_64-apple-ios-macabi' [-Werror,-Woverriding-t-option]
```
OTH, `jsoncpp/1.9.3` (also CMake-based) build fine, but with warning:
```
clang: warning: overriding '-mmacosx-version-min=13.0' option with '--target=x86_64-apple-ios-macabi' [-Woverriding-t-option]
```
as I understand flag `-mmacosx-version-min=13.0` is added by CMake itself, so it's upstream problem https://gitlab.kitware.com/cmake/cmake/-/issues/20132

also checked `libxml2/2.9.10` and `libiconv/1.16` (autotools-based), build just fine.

**Comment by SSE4** [comment]
okay, seems like this condition prevents `-mios-version-min` to be applied for `AutoToolsBuildEnvironment`:
https://github.com/conan-io/conan/blob/1d050595f809c0b8b073f4402509f6e1dca4bb38/conans/client/build/autotools_environment.py#L356
because `--target=x86_64-apple-ios-macabi` is already applied from the `architecture_flag`

**Comment by SSE4** [comment]
okay, one more thing. `zlib/1.2.11` (and several other recipes) needs the following code either removed or fixed:
https://github.com/conan-io/conan-center-index/blob/655abfca62804aeeee0804af7dd81b8a9ea15b86/recipes/zlib/1.2.11/conanfile.py#L69
as it doesn't pass `os.subsystem` and `os.sdk`, so it incorrectly computes flag `-mmacosx-version-min=13.0`

**Comment by memsharded** [comment]
Fixed in https://github.com/conan-io/conan/pull/8389, will be released in next patch 1.33.1
